map/search_mark.cpp

```diff
@@ -184,6 +196,23 @@ class SearchMarkTypeChecker
       {{"tourism", "museum"},        SearchMarkType::Museum},
       {{"amenity", "arts_centre"},   SearchMarkType::Art},
       {{"tourism", "gallery"},       SearchMarkType::Art},
+      {{"tourism", "attraction"},    SearchMarkType::Attraction},
+      {{"tourism", "viewpoint"},     SearchMarkType::Viewpoint},
+      {{"historic", "fort"},                        SearchMarkType::Remains},
+      {{"historic", "castle"},                      SearchMarkType::Remains},
+      {{"historic", "castle", "castrum"},           SearchMarkType::Remains},
+      {{"historic", "castle", "fortified_church"},  SearchMarkType::Remains},
+      {{"historic", "castle", "fortress"},          SearchMarkType::Remains},
+      {{"historic", "castle", "hillfort"},          SearchMarkType::Remains},
+      {{"historic", "castle", "kremlin"},           SearchMarkType::Remains},
+      {{"historic", "castle", "manor"},             SearchMarkType::Remains},
+      {{"historic", "castle", "palace"},            SearchMarkType::Remains},
+      {{"historic", "castle", "shiro"},             SearchMarkType::Remains},
+      {{"historic", "castle", "defensive"},         SearchMarkType::Remains},
+      {{"historic", "castle", "stately"},           SearchMarkType::Remains},
```

Comment:
Well, "remains" is not a good name for these types :)
"Castle" fits better here IMO.